homeassistant/components/localip/__init__.py

```diff
@@ -0,0 +1,12 @@
+"""Get the local IP (eth0 equivalent) of the Home Assistant instance."""
+import homeassistant.util as hass_util
+
+DOMAIN = "localip"
+
+
+async def async_setup(hass, config):
+    """Set up the interface with the detected IP."""
+    hass.states.async_set("localip.ip", hass_util.get_local_ip())
```

Comment:
This should probably be a sensor platform.

Reply:
I thought to do so, but wasn't sure of the semantics...  Basically just leave a docstring in __init__.py and move the rest to sensor.py?

Reply:
That works, but we prefer to centralize integrations with core logic in the component module and config under the integration domain. In this case there's not very much logic or config, so either way is fine, I think.

If you want to centralize you can load the platform via `discovery.async_load_platform`.

Reply:
Also, the util call does I/O so must be scheduled on the executor thread pool.

An easy way of doing this is to use the sync api of home assistant eg `def setup` instead of `async def async_setup`.

Reply:
I will split the platform - it's a good excuse for learning :)  If the sensor setup needs to use the sync API to keep the code in the correct threadpool, does that mean that I need to use `discovery.load_platform`, too?

Another similar question I had whether to leave it with a `configuration.yaml` config, or to use the new UI scaffolding.  It seemed to me a bit excessive to start with the new-style, but I'm happy to put in the time to do that too if you disagree.

Thanks for the super-fast turnaround!

Reply:
You can have the component set up be async and the platform set up be sync. It's only the current context that does the call that matters.

Eg

```py
# in __init__.py
async def async_setup(...):
    hass.async_create_task(
        async_load_platform(...)
    )
    return True


# in sensor.py
def setup_platform(...):
    local_ip = hass_util.get_local_ip()
```

Reply:
It's up to you if you want to implement config entry support with config flow or just use config yaml or both.

Reply:
Added in f127e21
I still need to add a test, and not sure if I ought to allow double-loading or not - can members give me educated feedback on that?  Thanks!